src/server/server.go

```diff
@@ -39,53 +41,57 @@ func (p *JSONRPCProxy) SelectHandler(method string) http.Handler {
 
 func (p *JSONRPCProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	body, err := io.ReadAll(r.Body)
+
 	if err != nil {
 		http.Error(w, "Bad request", http.StatusBadRequest)
 		return
 	}
+
 	rpcreq := medley.RPCRequest{}
 	err = json.Unmarshal(body, &rpcreq)
 	if err != nil {
 		http.Error(w, "Bad request", http.StatusBadRequest)
 		return
 	}
+
 	selectedHandler := p.SelectHandler(rpcreq.Method)
+
 	if selectedHandler == p.processor {
-		Logger.Info().Str("method", rpcreq.Method).Msg("dispatching to processor")
+		utils.Logger.Info().Str("method", rpcreq.Method).Msg("dispatching to processor")
 	} else {
-		Logger.Info().Str("method", rpcreq.Method).Msg("dispatching to backend")
+		utils.Logger.Info().Str("method", rpcreq.Method).Msg("dispatching to backend")
 	}
 
 	// make the body available again before letting reverse proxy handle the rest
 	r.Body = io.NopCloser(bytes.NewBuffer(body))
 	selectedHandler.ServeHTTP(w, r)
 }
 
-type Config struct {
-	BackendURL        *url.URL
-	HTTPListenAddress string
-}
-
 type server struct {
 	processor rpc.Processor
-	config    *Config
+	config    rpc.Config
+	sub       ethereum.Subscription
+	headers   chan *types.Header
 }
 
-func NewRPCService(processor rpc.Processor, config *Config) medleyService.Service {
+func NewRPCService(processor rpc.Processor, config rpc.Config, sub ethereum.Subscription, headers chan *types.Header) medleyService.Service {
 	return &server{
 		processor: processor,
 		config:    config,
+		sub:       sub,
+		headers:   headers,
 	}
 }
 
-func (srv *server) rpcHandler() (http.Handler, error) {
+func (srv *server) rpcHandler(ctx context.Context) (http.Handler, error) {
 	rpcServices := []rpc.RPCService{
 		&rpc.EthService{},
 	}
 
 	rpcServer := ethrpc.NewServer()
 	for _, service := range rpcServices {
-		service.InjectProcessor(srv.processor)
+		service.Init(srv.processor, srv.config, srv.sub, srv.headers)
+		service.HandleBlocks(ctx)
```

Comment:
Maybe I'm wrong here, but shouldn't this be run in its own go routine? Otherwise it will be blocking here forever.

Reply:
Yes, I've corrected it to run in its own routine.